File: src/main/webapp/js/service/stepTaskService.js
app.service('stepTaskService',function($http){

    // 任务步骤列表
    this.getAll=function(){
        return $http.get('../StepTaskController/getAll');
    }
    // 任务步骤
    this.getOne=function(taskId){
        return $http.get('../StepTaskController/getOne?taskId='+taskId);
    }
    // 待处理列表
    this.searchByUsername=function(username){
        return $http.get('../StepTaskController/searchByUsername?username='+username);
    }
    // 任务下的列表
    this.searchByProjectId=function(taskId){
        return $http.get('../StepTaskController/searchByTaskId?taskId='+taskId);
    }
    // 新建任务
    this.insert=function(entity){
        return $http.post('../StepTaskController/insert',entity);
    }
    // 删除用户
    this.dele=function(selectIds){
        return $http.get('../StepTaskController/delete?ids='+selectIds);
    }
    // 更新任务信息
    this.update=function(entity){
        return $http.post('../StepTaskController/update',entity);
    }
})

File: src/main/java/com/talenco/tasksystem/service/StepTaskService.java
package com.talenco.tasksystem.service;

import com.talenco.tasksystem.entity.Step;
import com.talenco.tasksystem.entity.StepTask;

import java.util.List;

public interface StepTaskService {

    List<StepTask> getAll();

    StepTask getOne(Long id);

    void insert(List<Step> steps,Long taskId);

    void update(StepTask stepTask);

    /**
     * 在删除任务task时,删除项目下的步骤stepTask,该功能不影响主要供能,可后续完成
     * @param ids 项目id数组
     */
    void delete(Long[] ids);

    List<StepTask> searchByTaskId(Long projectId);

    List<StepTask> searchByUsername(String username);
}
